Add a django_set_urlconf fixture (#1064)
pytest-dev/pytest-django: docs/helpers.rst

"fake-tests.example.com".

``django_set_urlconf``
~~~~~~~~~~~~~~~~~~~~~~

> This fixture allows setting the ``settings.ROOT_URLCONF`` to a value created within the test.

Reviewer: ```suggestion
Call this fixture to set ``settings.ROOT_URLCONF`` to a list of urlpatterns created within the test.
This keeps the URLs local unlike :ref:`pytest.mark.urls`.
```

(that reference probably needs adjusting to work)

Example
"""""""

::

    from django import http, urls

    def test_can_override_a_root_urlconf(django_set_urlconf):

>         def my_view(request: http.request.HttpRequest) -> http.response.HttpResponse:

Reviewer: drop the type hints, since most users don't use them
```suggestion
        def my_view(request):
```

            raise http.Http404()

>         my_urlpatterns = (urls.path("test/view", my_view, name="test_view"),)
>         django_set_urlconf(my_urlpatterns)

Reviewer: A list is more typical for `urlpatterns`, and using inline is maybe a little clearer?

```suggestion
        django_set_urlconf([
            urls.path("test/view", my_view, name="test_view")
        ])
```
